Remove old about activity and related resources

ravishanker opened the pull request:
Fixes #15803 

This PR removes the following unused files and updates their references since the new Unified About is already integrated into WPAndroid and Jetpack

- Removed AboutPreferenceCategory and related preferences from AppSettingsFragment
- deleted old AboutActivity and related resources
- deleted old LicencesActivity and related resources
- Clean up related strings and styles

To test:
- Launch the app
- On MySite, Go to Me by clicking on avatar on top right hand corner
- Click on App Settings, scroll down and notice there's no About the app section
- Return to Me by pressing back 
- Notice About WordPress (second from bottom) below Share WordPress with a friend, click on it to see new Unified About 


## Regression Notes
1. Potential unintended areas of impact
None as Unified About is already integrated

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Clean up unit tests

3. What automated tests I added (or what prevented me from doing so)
Not required

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

ravishanker committed:
Update AppSettingsFragment.java

Remove reference to AboutActivity

ravishanker committed:
Update AndroidManifest.xml

Remove AboutActivity entry in manifest

ravishanker committed:
Delete AboutActivity and Layout xml files

ravishanker committed:
Update Constants.java

Remove unused constant

ravishanker committed:
Update me_fragment.xml

Remove Recommend the app item

ravishanker committed:
Update MeFragment.kt

Remove references to Recommend the app item

ravishanker committed:
Update MeViewModel.kt

Remove references to Recommend the app logic

ravishanker committed:
Update MeViewModelTest.kt

Remove tests related to Recommend the app

ravishanker committed:
Update app_settings.xml

Remove about preference category

ravishanker committed:
Update key_strings.xml

Remove about pref related strings

ravishanker committed:
Update AppSettingsFragment.java

Remove references to About preference
Remove references to UnifiedAbout

ravishanker committed:
Delete LicensesActivity.java

It is moved UnifiedAbout Screen

ravishanker committed:
Update AndroidManifest.xml

Remove LicensesActivity entry

ravishanker committed:
Remove unused strings and styles

ravishanker committed:
fix ciktlint issues

renanferrari commented on WordPress/src/main/java/org/wordpress/android/ui/main/MeFragment.kt:
```diff
@@ -85,9 +77,7 @@ class MeFragment : Fragment(R.layout.me_fragment), OnScrollToTopListener {
     @Inject lateinit var meGravatarLoader: MeGravatarLoader
     @Inject lateinit var viewModelFactory: ViewModelProvider.Factory
     @Inject lateinit var mediaPickerLauncher: MediaPickerLauncher
-    @Inject lateinit var recommendTheAppFeatureConfig: RecommendTheAppFeatureConfig
     @Inject lateinit var sequencer: SnackbarSequencer
-    @Inject lateinit var unifiedAboutFeatureConfig: UnifiedAboutFeatureConfig
```
Since these are not being used anywhere anymore we can probably go ahead and remove them as well? I'd just check with the folks that worked on the "recommend the app" feature to double-check this is ok.

ravishanker replied:
Thank you @renanferrari , Removed them as well and pinged about Recommend The App.  Also moved this to 19.1

renanferrari reviewed (commented):
Thank you for the changes! I tested the app and can confirm it works as expected. That said, I'm not sure we should proceed with this right now. I'd wait a bit until we have a better understanding of how the feature is being used – our first data analysis should happen sometime next week. Makes sense?

I also left a comment below for your consideration.

ravishanker committed:
Delete RecommendTheAppFeatureConfig.kt

Removed unused feature config

ravishanker committed:
Delete UnifiedAboutFeatureConfig.kt

removed unused feature config

ravishanker commented:
Hi @develric - **Recommend The App** feature is moved to the new Unified About screen as **Share with friends** and has been replaced under Me with About WordPress that leads to new Unified About.  

Hence, I've removed the Recommend The App Feature config and its related resources from WPAndroid.  

Please let me know if that's OK and if I'm missing anything.  Thank you

develric commented:
Hey @ravishanker 👋 ; I haven't done a full code review since @renanferrari has already covered it. I tested the recommend the app feature and it worked well for me overall. A couple of notes from my side:

- If we remove the RecommendTheAppFeatureConfig I think we can remove also the RECOMMEND_THE_APP flag in build.gradle (similar for UnifiedAboutFeatureConfig)
- I see we now return a default share link (apps.wordpress.com) in case when the API fetch fails. I would consider keeping the campaign in the link so we can know the origin source for tracking purpose:
    -  For WordPress -> 'https://apps.wordpress.com/get?campaign=app_share_link'
    - For Jetpack -> 'https://jetpack.com/app?campaign=app_share_link'

I also see there is a somewhat related running PR [here](https://github.com/wordpress-mobile/WordPress-Android/pull/15818), so not sure about the plans for this one, let me know 🙇

ravishanker commented:
> I also see there is a somewhat related running PR [here](https://github.com/wordpress-mobile/WordPress-Android/pull/15818), so not sure about the plans for this one, let me know 🙇

Thank you @develric for taking a look.  Yes, we're restoring the feature.  So, I need to revert the related in this PR soon.

ravishanker committed:
Revert "Update me_fragment.xml"

This reverts commit 7c3aec43d8763405023c104d9c1f5dcf54f527b1.

ravishanker committed:
Revert "Update MeFragment.kt"

This reverts commit ffe4cf8e1e670c1603d00b77a91f9c2ac2fb3a45.

ravishanker committed:
revert recommend the app

ravishanker committed:
revert recommend the app related tests

ravishanker committed:
Revert "Delete RecommendTheAppFeatureConfig.kt"

This reverts commit 01951846de79d0b45d03dcba695e2f6d6f469cb6.

ravishanker committed:
restore share button string

ravishanker committed:
refactor unified about related

ravishanker committed:
Merge branch 'develop' into Remove-old-AboutActivity-and-clean-up

ravishanker committed:
keep the campaign in the share link

in case when the API fetch fails, keeping the campaign in the link so we can know the origin source for tracking purpose

renanferrari commented:
@ravishanker I took the liberty to change the milestone to `Future` here, since we don't know exactly when this is going to be merged.

Tiay8885 commented:
`Remove-old-AboutActivity-and-clean-up `

Tiay8885 reviewed (changes requested):
`Remove-old-AboutActivity-and-clean-up `

Tiay8885 reviewed (changes requested):
`Remove-old-AboutActivity-and-clean-up `

Tiay8885 commented:
`Remove-old-AboutActivity-and-clean-up `

Tiay8885 reviewed (approved):
`Remove-old-AboutActivity-and-clean-up `

Tiay8885 reviewed (changes requested):
`Remove-old-AboutActivity-and-clean-up `

ravishanker committed:
Merge branch 'trunk' into Remove-old-AboutActivity-and-clean-up

ravishanker committed:
remove unused string

irfano commented on WordPress/src/main/java/org/wordpress/android/ui/main/MeFragment.kt:
```diff
@@ -186,28 +198,6 @@ class MeFragment : Fragment(R.layout.me_fragment), OnScrollToTopListener {
-        if (unifiedAboutFeatureConfig.isEnabled()) {
-            aboutTheAppContainer.isVisible = true
-
-            if (BuildConfig.IS_JETPACK_APP) {
-                meAboutIcon.setImageResource(R.drawable.ic_jetpack_logo_white_24dp)
-            }
```
We shouldn't remove this to keep JP icon. Otherwise, the WP icon is shown on JP.
<img src="https://user-images.githubusercontent.com/2471769/209139956-297962ad-7728-475f-a667-72200012a902.png" height=640>

ravishanker replied:
Good catch 👍

irfano commented on WordPress/src/main/java/org/wordpress/android/util/config/UnifiedAboutFeatureConfig.kt:
```diff
@@ -1,19 +0,0 @@
-    appConfig: AppConfig
-) : FeatureConfig(
-        appConfig,
-        BuildConfig.UNIFIED_ABOUT,
```
We can also remove `UNIFIED_ABOUT` from `build.gradle`.

ravishanker replied:
Yes, 👍

irfano reviewed (changes requested):
The diff LGTM! Thank you for handling this. 👍🏻 
- What about renaming `UnifiedAboutActivity` to `AboutActivity`? Is it worth changing?

ravishanker committed:
Add back JP icon on Me screen

ravishanker committed:
Remove UNIFIED_ABOUT config field

ravishanker committed:
Fix lint issue function length over 60 lines

ravishanker committed:
Merge branch 'trunk' into Remove-old-AboutActivity-and-clean-up

ravishanker committed:
Merge branch 'trunk' into Remove-old-AboutActivity-and-clean-up

ravishanker commented:
>     * What about renaming `UnifiedAboutActivity` to `AboutActivity`? Is it worth changing?
It was intentional in naming it Unified prefix, since it was meant to be common across all the a8c apps!

irfano reviewed (approved):
LGTM! 👍🏻
